ecommerce/extensions/payment/processors/paystack.py

```diff
@@ -0,0 +1,200 @@
+""" Paystack payment processor. """
+from __future__ import absolute_import, unicode_literals
+
+import logging
+
+from django.urls import reverse
+from django.utils.functional import cached_property
+from oscar.apps.payment.exceptions import GatewayError
+from oscar.core.loading import get_model
+
+from ecommerce.core.url_utils import get_ecommerce_url
+from ecommerce.extensions.payment.exceptions import RefundError
+from ecommerce.extensions.payment.processors import BaseClientSidePaymentProcessor, HandledProcessorResponse
+from ecommerce.qverse_features.paystack.client import PaystackClient
+from ecommerce.qverse_features.paystack.constants import (
+    CREATE_REFUND_CODE,
+    FETCH_REFUND_CODE,
+    INITIALIZE_TRANSACTION_CODE
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Paystack(BaseClientSidePaymentProcessor):
+    NAME = 'paystack'
+
+    def __init__(self, site):
+        """
+        Constructs a new instance of the Paystack processor.
+
+        Raises:
+            KeyError: If no settings configured for this payment processor.
+        """
+        super(Paystack, self).__init__(site)
+        self.public_key =self.configuration['public_key']
+        self.secret_key = self.configuration['secret_key']
+
+    @cached_property
+    def paystack_client(self):
+        """
+        Returns paystack client instance with appropriate configuration
+        """
+        return PaystackClient(self.configuration['base_url'], self.secret_key)
+
+    @property
+    def cancel_url(self):
+        return get_ecommerce_url(self.configuration['cancel_checkout_path'])
+
+    def get_return_url(self):
```

Comment:
what if we make this a property as well? Just like the above? To increase consistency